packages/experience-builder-sdk/src/core/componentRegistry.ts

```diff
@@ -189,6 +192,115 @@ export const runRegisteredComponentValidations = () => {
   });
 };
 
+const getSingleCssVariableValue = (
+  element: HTMLDivElement,
+  cssVariableValue: string,
+  cssAttribute: any,
+) => {
+  element.style[cssAttribute] = cssVariableValue;
+  const styles = getComputedStyle(element);
+  const resolvedValue = styles.getPropertyValue(cssAttribute);
+  return resolvedValue;
+};
+
+const getAllCssVariableValues = (
+  element: HTMLDivElement,
+  cssVariable: Record<string, string>,
+  cssAttribute: any,
+) => {
+  const resolvedCssVariables = {} as Record<string, string>;
+
+  Object.keys(cssVariable).forEach((key) => {
+    const cssVariableValue = cssVariable[key];
+    if (CssVarRegex.test(cssVariableValue)) {
+      const resolvedValue = getSingleCssVariableValue(element, cssVariableValue, cssAttribute);
+      resolvedCssVariables[cssVariableValue] = resolvedValue;
+    }
+  });
+  return resolvedCssVariables;
+};
+
+const resolveCssVariables = (designTokensDefinition: DesignTokensDefinition) => {
+  const {
+    spacing,
+    sizing,
+    color,
+    borderRadius,
+    fontSize,
+    lineHeight,
+    letterSpacing,
+    textColor,
+    border,
+  } = designTokensDefinition;
+  const resolvedCssVariables = {} as Record<string, string>;
+
+  // Create an element
+  const element = document.createElement('div');
+  document.body.appendChild(element);
```

Comment:
Does this work with SSR? 🤔

Reply:
I would also discourage from trying to affecting the DOM from within a plain js function.

I think react would expect it to be done in the `useEffect`

Reply:
It is compatible with SSR, @Spring3 can you explain what the difference in using the useEffect does? I'm only temporarily creating it to grab the css values and then deleting the element.